cmd/resolutioncli/variable_source.go

```diff
@@ -19,12 +19,14 @@ package main
 import (
 	"github.com/operator-framework/deppy/pkg/deppy/input"
 
+	catalogclient "github.com/operator-framework/operator-controller/internal/catalogmetadata/client"
 	"github.com/operator-framework/operator-controller/internal/resolution/variablesources"
 )
 
-func newPackageVariableSource(packageName, packageVersion, packageChannel string) func(inputVariableSource input.VariableSource) (input.VariableSource, error) {
+func newPackageVariableSource(catalog catalogclient.CatalogClient, packageName, packageVersion, packageChannel string) func(inputVariableSource input.VariableSource) (input.VariableSource, error) {
```

Comment:
Global comments for all places we're referencing a CatalogClient in this PR:
1. Given the variable's type is CatalogClient, I would expect to see "client" somewhere in the variable's name
2. Maybe this is actually more like a "BundleProvider"? Or does that make it too abstract, and just calling this catalogClient would suffice?

Reply:
Personally I like `BundleProvider`, to me that makes it clear what the intended purpose of the type is.

Reply:
Eventually there might be a need to read other catalog info like channels and packages. So the best thing I came up with was "catalog client", but I don't really like it.

Happy to go with `BundleProvider` for now and we can rename later when/if we need to expand.